p2p/utils_test.go

```diff
@@ -0,0 +1,81 @@
+package p2p
+
+import (
+	"context"
+	"crypto/rand"
+	"strconv"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+
+	"github.com/lazyledger/optimint/config"
+	"github.com/lazyledger/optimint/log"
+)
+
+type testNet []*Client
+
+func (tn testNet) Start(ctx context.Context) (err error) {
+	for i := range tn {
+		err = multierr.Append(err, tn[i].Start(ctx))
```

Comment:
This won't append anything if the `Start` doesn't return an error, right?

Reply:
yes, it only appends if `err != nil`.